server/controller/src/main/java/ai/starwhale/mlops/datastore/DataStore.java

```diff
@@ -482,17 +627,7 @@ record = new HashMap<>();
                 }
                 records.removeIf(r -> r.record == null);
             }
-            var columnStatistics = new HashMap<String, ColumnStatistics>();
-            for (var table : tables) {
-                table.table.getColumnStatistics(table.columns)
-                        .forEach((k, v) -> columnStatistics.computeIfAbsent(k, x -> new ColumnStatistics()).merge(v));
-            }
-            return new RecordList(columnSchemaMap,
-                    columnStatistics.entrySet().stream()
-                            .collect(Collectors.toMap(Entry::getKey, entry -> entry.getValue().toColumnHintsDesc())),
-                    ret,
-                    (String) BaseValue.encode(lastKey, false, false),
-                    BaseValue.getColumnType(lastKey).name());
+            return resultResolver.apply(tables, columnSchemaMap, ret, lastKey);
```

Comment:
It does a lot of useless encoding when used by `scanKeyRange`.

Reply:
done